data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/processor/KinesisRecordProcessor.java

```diff
@@ -112,7 +114,18 @@ public KinesisRecordProcessor(final BufferAccumulator<Record<Event>> bufferAccum
     }
 
     private KinesisStreamConfig getStreamConfig(final KinesisSourceConfig kinesisSourceConfig) {
-        final Optional<KinesisStreamConfig> kinesisStreamConfig = kinesisSourceConfig.getStreams().stream().filter(streamConfig -> streamConfig.getName().equals(streamIdentifier.streamName())).findAny();
+        final Optional<KinesisStreamConfig> kinesisStreamConfig = kinesisSourceConfig.getStreams().stream().filter(streamConfig -> {
```

Comment:
We have a bit of split logic with how we ready the stream config and get the stream identifiers. I see other conditions based on the different options above in `KinesisClientApiHandler` and `KinesisMultiStreamTracker`.

There should be a good way to have a single class that can handle all the logic for determining the `KinesisStreamConfig`/`StreamConfig` for any stream or identifier. Maybe a class that maps all these on initialization?

Then you won't have to have all this split logic.